fingerprint/views.py

```diff
@@ -0,0 +1,102 @@
+from functools import wraps
+from django.views.decorators.csrf import csrf_exempt
+from django.views.generic import TemplateView
+from django.utils.decorators import method_decorator
+from django.http.response import HttpResponse
+from django.core.exceptions import BadRequest
+from django.utils.datastructures import MultiValueDictKeyError
+from datetime import timedelta
+
+from ipware import get_client_ip
+
+from .models import RequestFingerprint, BrowserFingerprint, UserSession
+
+
+def get_or_create_session_key(request) -> str:
+    if not request.session or not request.session.session_key:
+        request.session.create()
+    return request.session.session_key
+
+
+def fingerprint(fn):
+    """ A decorator which creates a backend Fingerprint object for the current request. """
+
+    @wraps(fn)
+    def wrapper(request, *args, **kwargs):
+        session_key = get_or_create_session_key(request)
+        RequestFingerprint.objects.create(
+            user_session=UserSession.objects.get_or_create(session_key=session_key)[0],
+            ip=get_client_ip(request)[0],
+            user_agent=request.META.get('HTTP_USER_AGENT', ''),
+            accept=request.META.get('HTTP_ACCEPT', ''),
+            content_encoding=request.META.get('HTTP_CONTENT_ENCODING', ''),
+            content_language=request.META.get('HTTP_CONTENT_LANGUAGE', ''),
```

Comment:
adding [:255] may prevent us from these exploding if someone comes with some funky values.
I have seen user agents containing more than 150+ characters.
Unlikely but unless we have something to lose here I would suggest saving someone from being woken by 500 in the production that don't really show any real issue.

Reply:
Good catch! For some reason I thought that django would auto-truncate, but it's not true. Thanks!